nexpose/nexpose_report.py

```diff
@@ -75,24 +70,371 @@ def __init__(self):
+
+        # TODO: draw the rest of the owl
+
+        return cfg
+
+    def __init__(self, name, template_id, format, id=-1, owner=None, timezone=None):
+        """
+
+        :param name: The name of the report configuration
+        :type name: str
+        :param template_id: The report template ID
+        :type template_id: str
+        :param format: The report output format (pdf|html|rtf|xml|text|csv|db|raw-xml|raw-xml-v2|ns-xml|qualys-xml|sql)
+        :type format: str
+        :param id: The report configuration ID, or -1 for a new configuration
+        :type id: int
+        :param owner: The user ID of the report owner
+        :type owner: int or None
+        :param timezone: The timezone of the report by name, e.g. America/Los_Angeles for Pacific (PDT/PST)
+        :type timezone: str
+        """
+        AdhocReportConfiguration.__init__(self, template_id, format, None, owner, timezone)
+        self.name = name
+        self.id = id
         self.users = []
         self.generate = None
-        self.delivery = ''
-        self.dbexport = ''
-        self.credentials = ''
-        self.parameter_name = ''  # TODO: ??
+        self.frequency = None
+        self.delivery = None
+        self.dbexport = None  # TODO: needs DBExport class implemented
+        self.credentials = None  # TODO: needs ExportCredential class implemented
+
+    def AsXML(self, exclude_id=False):
+        attributes = {'id': self.id, 'name': self.name, 'template-id': self.template_id, 'format': self.format}
+        if self.owner:
+            attributes['owner'] = self.owner
+        if self.timezone:
+            attributes['timezone'] = self.timezone
+
+        xml_data = create_element('ReportConfig', attributes)
+
+        xml_filters = create_element('Filters')
+        for report_filter in self.filters:
+            xml_filters.append(report_filter.AsXML())
+
+        xml_data.append(xml_filters)
+
+        if self.baseline:
+            xml_baseline = create_element('Baseline', {'compareTo': self.baseline})
+        else:
+            xml_baseline = create_element('Baseline')
+
+        xml_data.append(xml_baseline)
+
+        xml_users = create_element('Users')
+        for user in self.users:
+            xml_users.append(create_element('user', {'id': user}))
+        xml_data.append(xml_users)
+
+        if self.frequency:
+            xml_data.append(self.frequency.AsXML())
+
+        if self.delivery:
+            xml_data.append(self.delivery.AsXML())
+
+        if self.dbexport:
+            pass  # TODO needs DBExport class implemented
+
+        return xml_data
+
+
+class Filter:
+    def __init__(self, type, id):
+        """
+        The type can be one of:
+
+        - site
+        - group
+        - device
+        - tag
+        - scan
+        - vuln-categories
+        - vuln-severity
+        - vuln-status
+        - cyberscope-component
+        - cyberscope-bureau
+        - cyberscope-enclave
```

Comment:
Does this work for SQL reports?
I added Adhoc Reports to the Nexpose Komand plugin, and I needed to add `version` and `query` to this list.

Reply:
Yep just need to update the comment here.